### PR TITLE
feat(api): add `ledger/updates/by-milestone` endpoint

### LINKED ISSUE #261: Split cursor string on `.` and use Bee types to validate

### LINKED ISSUE #315: Add an endpoint for ledger updates per milestone
We should add an endpoint that allows the user to retrieve all ledger updates for a given milestone based on the `LedgerUpdate` collection.

This endpoint should be part of the `history/v2` API and also be reflected in the TIP (https://editor.swagger.io/?url=https://raw.githubusercontent.com/iotaledger/tips/explorer-api/tips/TIP-0036/tx-history-rest-api.yaml) as well.

It should probably be something like: `api/history/v2/ledger/updates/by-milestone/{milestoneId}`

### PR DESCRIPTION
**Pull request by Alex6323**
Closes #315
Closes #261

### CODE REVIEW

File: bin/inx-chronicle/src/api/stardust/history/routes.rs [hunk 1]
```diff
@@ -66,9 +68,54 @@ async fn transaction_history(
...
+    Ok(TransactionsPerAddressResponse {
         address,
+        items: transactions,
+        cursor,
+    })
+}
+
+async fn transactions_by_milestone_history(
+    database: Extension<MongoDb>,
+    Path(milestone_id): Path<String>,
```
Comment (grtlr) [inline bin/inx-chronicle/src/api/stardust/history/routes.rs:80]: Sorry, I made a mistake here, this should actually be `milestone_index`, which we use throughout most of the API.

File: bin/inx-chronicle/src/api/stardust/history/extractors.rs [hunk 2]
```diff
@@ -11,49 +11,88 @@ use serde::Deserialize;
...
+        if let Some(cursor) = cursor {
+            // Unwrap: Infallable as long as the regex is valid
+            let regex = Regex::new(BY_MILESTONE_HISTORY_CURSOR_REGEX).unwrap();
+            let captures = regex.captures(&cursor).ok_or(ParseError::BadPagingState)?;
+            start_output_id
+                .replace(OutputId::from_str(captures.get(1).unwrap().as_str()).map_err(ApiError::bad_parse)?);
+            page_size.replace(captures.get(2).unwrap().as_str().parse().map_err(ApiError::bad_parse)?);
+        }
+        Ok(HistoryByMilestonePagination {
+            page_size: page_size.unwrap_or(100),
```
[the remark this answers is not in the document]
Comment (Alex6323) [reply bin/inx-chronicle/src/api/stardust/history/extractors.rs:94]: Done

File: bin/inx-chronicle/src/api/stardust/history/extractors.rs [hunk 3]
```diff
@@ -11,49 +11,88 @@ use serde::Deserialize;
 
 use crate::api::{error::ParseError, ApiError};
 
-const HISTORY_PAGING_REGEX: &str = r"^([0-9]+)\.(0x(?:[0-9a-fA-F]{2})+)\.([0-9]+)$";
+const BY_ADDRESS_HISTORY_CURSOR_REGEX: &str = r"^([0-9]+)\.(0x(?:[0-9a-fA-F]{2})+)\.([0-9]+)$";
+const BY_MILESTONE_HISTORY_CURSOR_REGEX: &str = r"^(0x(?:[0-9a-fA-F]{2})+)\.([0-9]+)$";
```
[the remark this answers is not in the document]
Comment (Alex6323) [reply bin/inx-chronicle/src/api/stardust/history/extractors.rs]: Done

File: bin/inx-chronicle/src/api/stardust/history/routes.rs [hunk 4]
```diff
@@ -66,9 +68,54 @@ async fn transaction_history(
...
+
+    let transactions = records
+        .into_iter()
+        .map(|rec| {
+            Ok(Transfer {
+                output_id: rec.output_id.to_hex(),
+                is_spent: rec.is_spent,
+                milestone_index: rec.at.milestone_index,
+                milestone_timestamp: rec.at.milestone_timestamp,
+            })
```
[the remark this answers is not in the document]
Comment (Alex6323) [reply bin/inx-chronicle/src/api/stardust/history/routes.rs]: Done

File: bin/inx-chronicle/src/api/stardust/history/routes.rs (hunk 1 above)
Comment (Alex6323) [reply bin/inx-chronicle/src/api/stardust/history/routes.rs]: Changed.

File: src/db/collections/milestone.rs [hunk 5]
```diff
@@ -140,7 +140,28 @@ impl MongoDb {
...
+            )
+            .await?
+            .try_next()
+            .await?
+            .map(bson::from_document)
+            .transpose()?;
+
+        Ok(milestone_index)
+    }
+
```
[the remark this answers is not in the document]
Comment (Alex6323) [reply src/db/collections/milestone.rs]: Removed.

**Pull request merged**